### ISSUE TITLE
Is it possible to use this with mutable fragment lists with view pager 2?

### ISSUE THREAD

**Issue by kaushalyap** [tommybuonomo/dotsindicator#58]
Is it possible to use this with mutable fragment lists with view pager 2? Does number of dots automatically get updated when we remove/add fragments?

**Comment by tommybuonomo** [comment]
Hi @kaushalyap , as long as you have a ViewPager2, you can

**Comment by kaushalyap** [comment]
I thought it auto update # dots when # fragments changes, currently I set the viewpager again when # fragments changes

ex:

```
binding.btnNext.setOnClickListener {
            viewPagerAdapter.removeFragment(2)
            binding.dotsIndicator.setViewPager2(binding.vpHabitSteps)
} 
```

Can you guys make this automatic? (# dots adjust to # fragments)

**Comment by tommybuonomo** [comment]
What's inside your `removeFragment ()` method ? Maybe you forgot to call `notifyDataSetChanged` inside right ?
The refresh is automatic, you don't have to set the ViewPager again if you notify your adapter that the items changed

**Comment by kaushalyap** [comment]
I have following
```
 fun removeFragment(position: Int) {
        fragmentList.removeAt(position)
        notifyItemRemoved(position)
 }
```

It does work(automatically update # dots) with `notifyDataSetChanged`, But  does not work with `notifyItemRemoved(position)` why does it does not work with `notifyItemRemoved(position)`?

**Comment by kaushalyap** [comment]
paging @tommybuonomo

**Comment by vkotovv** [comment]
For some reason, if I remove the first item from my ViewPager2, the dots count is updated, but the indicator' selected item is not updated (no dots are selected). What could be the issue? I'm using version 4.1.2

**Comment by vkotovv** [comment]
Related issue - #73

**Comment by tommybuonomo** [comment]
Hello, thanks for your contribution !
This issue is now fixed in the new version `4.3`
Thanks 🔥
